File: clang/test/CodeGen/split-lto-unit.c
// ; Check that -flto=thin without -fsplit-lto-unit has EnableSplitLTOUnit = 0
// RUN: %clang_cc1 -flto=thin -emit-llvm-bc < %s | llvm-dis -o - | FileCheck %s
// RUN: %clang_cc1 -flto=thin -emit-llvm < %s | FileCheck %s
// RUN: %clang_cc1 -flto=thin -funified-lto -triple=x86_64-scei-ps4 -emit-llvm-bc < %s | llvm-dis -o - | FileCheck %s
// CHECK: !{i32 1, !"EnableSplitLTOUnit", i32 0}
//
// ; Check that -flto=thin with -fsplit-lto-unit has EnableSplitLTOUnit = 1
// RUN: %clang_cc1 -flto=thin -fsplit-lto-unit -emit-llvm-bc < %s | llvm-dis -o - | FileCheck %s --check-prefix=SPLIT
// RUN: %clang_cc1 -flto=thin -fsplit-lto-unit -emit-llvm < %s | FileCheck %s --check-prefix=SPLIT
// SPLIT: !{i32 1, !"EnableSplitLTOUnit", i32 1}
//
// ; Check that regular LTO has EnableSplitLTOUnit = 1
// RUN: %clang_cc1 -flto -triple x86_64-pc-linux-gnu -emit-llvm-bc < %s | llvm-dis -o - | FileCheck %s --implicit-check-not="EnableSplitLTOUnit" --check-prefix=SPLIT
// RUN: %clang_cc1 -flto -triple x86_64-pc-linux-gnu -emit-llvm < %s | FileCheck %s --implicit-check-not="EnableSplitLTOUnit" --check-prefix=SPLIT

// ; Check that regular LTO has EnableSplitLTOUnit = 1, if using distinct pipelines. For unified pipelines, use EnableSplitLTOUnit = 0.
// RUN: %clang_cc1 -flto -triple x86_64-pc-linux-gnu -emit-llvm-bc < %s | llvm-dis -o - | FileCheck %s --implicit-check-not="EnableSplitLTOUnit" --check-prefix=SPLIT
// RUN: %clang_cc1 -flto -funified-lto -triple x86_64-scei-ps4 -emit-llvm-bc < %s | llvm-dis -o - | FileCheck %s
// RUN: %clang_cc1 -flto=thin -funified-lto -fsplit-lto-unit -triple x86_64-pc-linux-gnu -emit-llvm-bc < %s | llvm-dis -o - | FileCheck %s --implicit-check-not="EnableSplitLTOUnit" --check-prefix=SPLIT

int main(void) {}

File: flang/test/Driver/split-lto-unit.f90
! Check that -fsplit-lto-unit is passed to fc1 by the driver with -fsplit-lto-unit
! RUN: %flang -c -### -fsplit-lto-unit %s 2>&1 | FileCheck %s --check-prefix=SPLIT

! Check that -fsplit-lto-unit is passed to fc1 by the driver with -fsplit-lto-unit
! RUN: %flang -c -### -fno-split-lto-unit -fsplit-lto-unit %s 2>&1 | FileCheck %s  --check-prefix=SPLIT

! Check that -fsplit-lto-unit is not passed to fc1 by the driver with -fno-split-lto-unit
! RUN: %flang -c -### -fno-split-lto-unit %s 2>&1 | FileCheck %s  --check-prefix=NO-SPLIT

! Check that -fsplit-lto-unit is not passed to fc1 by the driver with -fno-split-lto-unit
! RUN: %flang -c -### -fsplit-lto-unit -fno-split-lto-unit %s 2>&1 | FileCheck %s  --check-prefix=NO-SPLIT

! Check that the driver does not pass -fsplit-lto-unit to fc1 by default
! RUN: %flang -c -### %s 2>&1 | FileCheck %s  --check-prefix=NO-SPLIT
! RUN: %flang -c -### -flto %s 2>&1 | FileCheck %s  --check-prefix=NO-SPLIT
! RUN: %flang -c -### -flto=thin %s 2>&1 | FileCheck %s  --check-prefix=NO-SPLIT

! SPLIT: "-fc1"
! SPLIT-SAME: "-fsplit-lto-unit"
! NO-SPLIT-NOT: "-fsplit-lto-unit"

program main
end program main
